common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -342,16 +360,32 @@ def from_storable(self, edit_info):
         self.original_usage = edit_info.get('original_usage', None)
         self.original_usage_version = edit_info.get('original_usage_version', None)
 
-    def __str__(self):
+    def get_source_version(self, value_if_unset=UNSET):
+        """
+        Return the version of the given database representation of a block.
+        """
+        if value_if_unset is UNSET:
+            # Parameter was not passed-in.
+            value_if_unset = self.update_version
+        if self.source_version is not UNSET:
+            return self.source_version
+        else:
+            return value_if_unset
+
+    def __repr__(self):
+        # pylint: disable=bad-continuation
         return ("EditInfo(previous_version={0.previous_version}, "
                 "update_version={0.update_version}, "
-                "source_version={0.source_version}, "
+                "source_version={1}, "
                 "edited_on={0.edited_on}, "
                 "edited_by={0.edited_by}, "
                 "original_usage={0.original_usage}, "
                 "original_usage_version={0.original_usage_version}, "
                 "_subtree_edited_on={0._subtree_edited_on}, "
-                "_subtree_edited_by={0._subtree_edited_by})").format(self)
+                "_subtree_edited_by={0._subtree_edited_by})").format(
```

Comment:
Nit: `EditInfo` should be written as `0.__class__.__name__`, so that it covers subclasses correctly.